src/singletons/NativeMessaging.cpp

```diff
@@ -60,6 +60,18 @@ void registerNmHost(const Paths &paths)
         auto obj = getBaseDocument();
         QJsonArray allowedOriginsArr = {
             u"chrome-extension://%1/"_s.arg(EXTENSION_ID)};
+
+        // Parse additional extension IDs from settings
+        QString additionalIDs = getSettings()->additionalExtensionIDs.getValue();
+        if (!additionalIDs.isEmpty())
+        {
+            QStringList idList = additionalIDs.split(';');
+            for (const QString &id : idList)
+            {
+                allowedOriginsArr.append(u"chrome-extension://%1/"_s.arg(id));
+            }
```

Comment:
We should trim the IDs and ignore empty strings (so `foo; bar;baz;` becomes `foo`, `bar`, `baz`). See [`QString::trimmed`](https://doc.qt.io/qt-6/qstring.html#trimmed) and [`QString::isEmpty`](https://doc.qt.io/qt-6/qstring.html#isEmpty).